I guess https://github.com/MarlinFirmware/Marlin/blob/9d1d35f1ed07a8884b928e9ee18f06e5f84f28de/buildroot/share/PlatformIO/scripts/generic_create_variant.py#L31 needs to be changed to `usb-host-msc-cdc-msc-2` too.
With this change I get the following compile error:
```
framework-arduinoststm32\cores\arduino\stm32\usb\usbd_conf.c: In function 'USBD_LL_Init':
framework-arduinoststm32\cores\arduino\stm32\usb\usbd_conf.c:532:25: error: 'ep_dep' undeclared (first use in this function); did you mean 'ep_def'?
  532 |     ep_desc_t *pDesc = &ep_dep[i];
      |                         ^~~~~~
      |                         ep_def
```
I am using the env STM32F103RC_btt_USB and BOARD_BTT_SKR_E3_DIP as board. Let me know if you require more configuration info or need help testing.

Thanks!

--- a/Marlin/src/HAL/STM32/inc/Conditionals_adv.h
+++ b/Marlin/src/HAL/STM32/inc/Conditionals_adv.h
@@ -21,7 +21,7 @@
  */
 #pragma once
 
-#if defined(USBD_USE_CDC_MSC) && DISABLED(NO_SD_HOST_DRIVE)
+#if BOTH(SDSUPPORT, USBD_USE_CDC_MSC) && DISABLED(NO_SD_HOST_DRIVE)
   #define HAS_SD_HOST_DRIVE 1
 #endif
 

--- a/buildroot/share/PlatformIO/scripts/generic_create_variant.py
+++ b/buildroot/share/PlatformIO/scripts/generic_create_variant.py
@@ -28,7 +28,7 @@ if len(platform_packages) == 0:
 else:
     platform_name = PackageSpec(platform_packages[0]).name
 
-if platform_name in [ "usb-host-msc", "usb-host-msc-cdc-msc", "tool-stm32duino" ]:
+if platform_name in [ "usb-host-msc", "usb-host-msc-cdc-msc", "usb-host-msc-cdc-msc-2", "tool-stm32duino" ]:
     platform_name = "framework-arduinoststm32"
 
 FRAMEWORK_DIR = platform.get_package_dir(platform_name)

--- a/ini/stm32f1.ini
+++ b/ini/stm32f1.ini
@@ -69,7 +69,7 @@ board_upload.offset_address = 0x08007000
 [env:STM32F103RC_btt_USB]
 extends           = env:STM32F103RC_btt
 platform          = ${common_stm32.platform}
-platform_packages = framework-arduinoststm32@https://github.com/rhapsodyv/Arduino_Core_STM32/archive/refs/heads/usb-host-msc-cdc-msc-2.zip
+platform_packages = framework-arduinoststm32@https://github.com/rhapsodyv/Arduino_Core_STM32/archive/usb-host-msc-cdc-msc-2.zip
 build_unflags     = ${common_stm32.build_unflags} -DUSBD_USE_CDC
 build_flags       = ${env:STM32F103RC_btt.build_flags} ${env:stm32_flash_drive.build_flags}
   -DUSBCON

--- a/ini/stm32f4.ini
+++ b/ini/stm32f4.ini
@@ -146,7 +146,7 @@ debug_init_break  =
 # USB Flash Drive mix-ins for STM32
 #
 [stm_flash_drive]
-platform_packages = framework-arduinoststm32@https://github.com/rhapsodyv/Arduino_Core_STM32/archive/refs/heads/usb-host-msc-cdc-msc-2.zip
+platform_packages = framework-arduinoststm32@https://github.com/rhapsodyv/Arduino_Core_STM32/archive/usb-host-msc-cdc-msc-2.zip
 build_flags       = ${common_stm32.build_flags}
   -DHAL_PCD_MODULE_ENABLED -DHAL_HCD_MODULE_ENABLED
   -DUSBHOST -DUSBH_IRQ_PRIO=3 -DUSBH_IRQ_SUBPRIO=4
@@ -424,7 +424,7 @@ build_flags       = ${stm_flash_drive.build_flags} ${stm32f4_I2C1.build_flags}
 [env:mks_robin_nano_v3_usb_flash_drive_msc]
 platform          = ${common_stm32.platform}
 extends           = env:mks_robin_nano_v3
-platform_packages = framework-arduinoststm32@https://github.com/rhapsodyv/Arduino_Core_STM32/archive/refs/heads/usb-host-msc-cdc-msc-2.zip
+platform_packages = framework-arduinoststm32@https://github.com/rhapsodyv/Arduino_Core_STM32/archive/usb-host-msc-cdc-msc-2.zip
 build_unflags     = ${common_stm32.build_unflags} -DUSBD_USE_CDC
 build_flags       = ${stm_flash_drive.build_flags} ${stm32f4_I2C1.build_flags}
   -DUSBCON

